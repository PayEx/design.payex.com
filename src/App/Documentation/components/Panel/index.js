import React from "react";
import PrismCode from "react-prism";

import { ComponentPreview, DocContainer, Property } from "#";
import PanelComponent from "@/Panel";

const bodyContent = [
    "Your main panel content is put here.",
    "Some more content over here.",
    "And one more line, just to be safe."
];

const TableContent = () => (
    <table className="table">
        <thead>
            <tr>
                <th scope="col">#</th>
                <th scope="col">First Name</th>
                <th scope="col">Last Name</th>
                <th scope="col">Location</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <th scope="row">1</th>
                <td>Sven</td>
                <td>Svensson</td>
                <td>Visby</td>
            </tr>
            <tr>
                <th scope="row">2</th>
                <td>Sara</td>
                <td>Svensson</td>
                <td>Stockholm</td>
            </tr>
            <tr>
                <th scope="row">3</th>
                <td>Ola</td>
                <td>Nordmann</td>
                <td>Oslo</td>
            </tr>
            <tr>
                <th scope="row">4</th>
                <td>Holger</td>
                <td>Danske</td>
                <td>Copenhagen</td>
            </tr>
            <tr>
                <th scope="row">5</th>
                <td>Matti</td>
                <td>Meikäläinen</td>
                <td>Lahti</td>
            </tr>
        </tbody>
    </table>
);

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>
            A panel consists of three parts: A header, a body and a footer. You can mostly put whatever you want in any of these,
            but try not to put too much in the header and footer as these are meant to contain short and concise information about the body content.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <PanelComponent
                title="Panel title"
                subTitle="Panel subtitle"
                footerText="Footer content goes here."
                footerBtnText="Footer button"
                bodyContent={bodyContent}
            />
        </ComponentPreview>
    </>
);

const PanelHeaders = () => (
    <>
        <h2 id="panel-headers">Panel headers</h2>
        <p>
            You can pick between two different panel headers, brand color with white text and white with green text. add <Property value=".panel-default" /> or <Property value=".panel-brand" /> to
            the header to use one of them.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <PanelComponent title="Default panel" />{"\n\n"}
            <PanelComponent type="brand" title="Panel brand" />
        </ComponentPreview>
    </>
);

const PanelDark = () => (
    <>
        <h2 id="panel-dark-mode">Panel dark mode</h2>
        <p>Dark body with <Property value=".panel-dark" />...</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <PanelComponent
                type="brand"
                title="Panel title"
                subTitle="Panel subtitle"
                bodyContent={bodyContent}
                footerText="Footer content goes here."
                footerBtnText="Footer button"
                darkMode
            />
        </ComponentPreview>
    </>
);

const PanelMuted = () => (
    <>
        <h2 id="muted-panel">Muted panel</h2>
        <p>
            If you wish to use a completely gray panel you can add <Property value=".panel-muted" /> to the panel <PrismCode className="language-html">{"<section>"}</PrismCode>.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <PanelComponent
                type="muted"
                title="Panel title"
                subTitle="Panel subtitle"
                bodyContent={bodyContent}
                footerText="Footer content goes here."
                footerBtnText="Footer button"
            />
        </ComponentPreview>
    </>
);

const PanelTable = () => (
    <>
        <h2 id="panel-with-table">Displaying a full width table</h2>
        <p>
            To display a full width table in a panel use <Property value=".panel-table" /> on a div wrapping the table.
            Put <Property value=".panel-table" /> as a direct child of <Property value=".panel" />. Tables put in a panel are given a
            border bottom to seperate it from the following elements.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <PanelComponent
                type="brand"
                title="Panel title"
                subTitle="Panel subtitle"
                bodyContent={bodyContent}
                footerText="Footer content goes here."
                footerBtnText="Footer button"
                panelTable={TableContent}
            />
        </ComponentPreview>
    </>
);

const Panel = () => (
    <DocContainer docToc>
        <p className="lead">
            Panels are large containers meant to be able to contain all of our other components.
        </p>
        <Overview />
        <PanelHeaders />
        <PanelDark />
        <PanelMuted />
        <PanelTable />
    </DocContainer>
);

export default Panel;

/* For testing */
export { Overview, PanelHeaders, PanelDark, PanelMuted, PanelTable, TableContent };
