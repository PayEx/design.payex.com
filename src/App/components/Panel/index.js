import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const Panel = ({ id, type, title, subTitle, removeBottomPadding, removeAllPadding, halfPadding, darkMode, children, panelTable, footerText }) => {
    const panelClasses = classnames(
        "panel",
        type ? `panel-${type}` : "panel-default",
        removeBottomPadding ? "panel-bottomless" : null,
        removeAllPadding ? "panel-no-padding" : null,
        halfPadding ? "panel-half-padding" : null,
        darkMode ? "panel-dark" : null
    );

    return (
        <section
            className={panelClasses}
            id={id}>
            {title
                ? (<header className="panel-header">
                    <h2 className="panel-title">{title}</h2>
                    {subTitle ? <p className="panel-sub-title">{subTitle}</p> : null}
                </header>)
                : null}
            {children
                ? (<div className="panel-body">
                    {children}
                </div>)
                : null}
            {panelTable
                ? (<div className="panel-table">
                    {panelTable}
                </div>)
                : null}
            {footerText
                ? (<footer className="panel-footer">
                    <p>{footerText}</p>
                </footer>)
                : null}
        </section>
    );
};

Panel.propTypes = {
    title: PropTypes.string,
    headerClass: PropTypes.bool,
    subTitle: PropTypes.string,
    id: PropTypes.string,
    type: PropTypes.oneOf(["brand", "default", "muted", ""]),
    halfPadding: PropTypes.bool,
    removeAllPadding: PropTypes.bool,
    removeBottomPadding: PropTypes.bool,
    darkMode: PropTypes.bool,
    bodyContent: PropTypes.object,
    footerClass: PropTypes.bool,
    footerText: PropTypes.string
};

export default Panel;
