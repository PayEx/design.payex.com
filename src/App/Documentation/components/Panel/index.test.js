import React from "react";
import { shallow } from "enzyme";

import Panel, { Overview, PanelHeaders, PanelDark, PanelMuted, PanelTable } from "./index";

describe("Documentation: Panel", () => {
    it("is defined", () => {
        expect(Panel).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Panel />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("PanelHeaders", () => {
        it("is defined", () => {
            expect(PanelHeaders).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<PanelHeaders />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("PanelDark", () => {
        it("is defined", () => {
            expect(PanelDark).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<PanelDark />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("PanelMuted", () => {
        it("is defined", () => {
            expect(PanelMuted).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<PanelMuted />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("PanelTable", () => {
        it("is defined", () => {
            expect(PanelTable).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<PanelTable />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
