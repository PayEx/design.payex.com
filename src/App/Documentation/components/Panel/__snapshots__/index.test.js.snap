// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: Panel Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    A panel consists of three parts: A header, a body and a footer. You can mostly put whatever you want in any of these, but try not to put too much in the header and footer as these are meant to contain short and concise information about the body content.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Panel
      footerBtnText="Footer button"
      footerText="Footer content goes here."
      subTitle="Panel subtitle"
      title="Panel title"
    >
      <p>
        Your main panel content is put here.
      </p>
      <p>
        Some more content over here.
      </p>
      <p>
        And one more line, just to be safe.
      </p>
    </Panel>
  </ComponentPreview>
  <Alert
    icon="warning"
    type="warning"
  >
    <h4>
      Deprecation warning
    </h4>
    <p>
      Only using 
      <PrismCode
        className="language-html"
        component="code"
      >
        &lt;header&gt;
      </PrismCode>
       and 
      <PrismCode
        className="language-html"
        component="code"
      >
        &lt;footer&gt;
      </PrismCode>
       is deprecated. Add 
      <Property
        value=".panel-header"
      />
       or 
      <Property
        value=".panel-footer"
      />
       to your panel header/footer element.
    </p>
  </Alert>
</Fragment>
`;

exports[`Documentation: Panel PanelDark renders 1`] = `
<Fragment>
  <h2
    id="panel-dark-mode"
  >
    Panel dark mode
  </h2>
  <p>
    Dark body with 
    <Property
      value=".panel-dark"
    />
    ...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Panel
      darkMode={true}
      footerBtnText="Footer button"
      footerText="Footer content goes here."
      subTitle="Panel subtitle"
      title="Panel title"
      type="brand"
    >
      <p>
        Your main panel content is put here.
      </p>
      <p>
        Some more content over here.
      </p>
      <p>
        And one more line, just to be safe.
      </p>
    </Panel>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Panel PanelHeaders renders 1`] = `
<Fragment>
  <h2
    id="panel-headers"
  >
    Panel headers
  </h2>
  <p>
    You can pick between two different panel headers, brand color with white text and white with green text. Add 
    <Property
      value=".panel-default"
    />
     or 
    <Property
      value=".panel-brand"
    />
     to the header to use one of them.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Panel
      title="Default panel"
    />
    


    <Panel
      title="Panel brand"
      type="brand"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Panel PanelMuted renders 1`] = `
<Fragment>
  <h2
    id="muted-panel"
  >
    Muted panel
  </h2>
  <p>
    If you wish to use a completely gray panel you can add 
    <Property
      value=".panel-muted"
    />
     to the panel 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;section&gt;
    </PrismCode>
    .
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Panel
      footerBtnText="Footer button"
      footerText="Footer content goes here."
      subTitle="Panel subtitle"
      title="Panel title"
      type="muted"
    >
      <p>
        Your main panel content is put here.
      </p>
      <p>
        Some more content over here.
      </p>
      <p>
        And one more line, just to be safe.
      </p>
    </Panel>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Panel PanelTable renders 1`] = `
<Fragment>
  <h2
    id="panel-with-table"
  >
    Displaying a full width table
  </h2>
  <p>
    To display a full width table in a panel use 
    <Property
      value=".panel-table"
    />
     on a div wrapping the table. Put 
    <Property
      value=".panel-table"
    />
     as a direct child of 
    <Property
      value=".panel"
    />
    . Tables put in a panel are given a border bottom to seperate it from the following elements.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Panel
      footerBtnText="Footer button"
      footerText="Footer content goes here."
      panelTable={
        <table
          className="table"
        >
          <thead>
            <tr>
              <th
                scope="col"
              >
                #
              </th>
              <th
                scope="col"
              >
                First Name
              </th>
              <th
                scope="col"
              >
                Last Name
              </th>
              <th
                scope="col"
              >
                Location
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th
                scope="row"
              >
                1
              </th>
              <td>
                Sven
              </td>
              <td>
                Svensson
              </td>
              <td>
                Visby
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                2
              </th>
              <td>
                Sara
              </td>
              <td>
                Svensson
              </td>
              <td>
                Stockholm
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                3
              </th>
              <td>
                Ola
              </td>
              <td>
                Nordmann
              </td>
              <td>
                Oslo
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                4
              </th>
              <td>
                Holger
              </td>
              <td>
                Danske
              </td>
              <td>
                Copenhagen
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                5
              </th>
              <td>
                Matti
              </td>
              <td>
                Meikäläinen
              </td>
              <td>
                Lahti
              </td>
            </tr>
          </tbody>
        </table>
      }
      subTitle="Panel subtitle"
      title="Panel title"
      type="brand"
    >
      <p>
        Your main panel content is put here.
      </p>
      <p>
        Some more content over here.
      </p>
      <p>
        And one more line, just to be safe.
      </p>
    </Panel>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Panel renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Panels are large containers meant to be able to contain all of our other components.
  </p>
  <Overview />
  <PanelHeaders />
  <PanelDark />
  <PanelMuted />
  <PanelTable />
</DocContainer>
`;
